apps/internal/base/base.go

```diff
@@ -345,6 +350,24 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
 		if err == nil {
+			if shouldRefresh(storageTokenResponse.AccessToken.RefreshOn.T) {
+				if tr, er := b.Token.Credential(ctx, authParams, silent.Credential); er == nil {
+					return b.AuthResultFromToken(ctx, authParams, tr)
+				} else if callErr, ok := er.(*errors.CallErr); ok {
+					// Check if the error is of type CallErr and matches the relevant status codes
```

Comment:
this says literally what the code does but not why the code does it, which is the interesting and perhaps mysterious part. Consider commenting instead in the empty case below with something like "refreshing failed with a non-fatal error; return the cached token"

Reply:
Updated